miniapp/miniapp_triangular_multiplication.cpp

```diff
@@ -0,0 +1,281 @@
+      : MiniappOptions(vm), m(vm["m"].as<SizeType>()), n(vm["n"].as<SizeType>()),
+        mb(vm["mb"].as<SizeType>()), nb(vm["nb"].as<SizeType>()),
+        side(dlaf::miniapp::parseSide(vm["side"].as<std::string>())),
+        uplo(dlaf::miniapp::parseUplo(vm["uplo"].as<std::string>())),
+        op(dlaf::miniapp::parseOp(vm["op"].as<std::string>())),
+        diag(dlaf::miniapp::parseDiag(vm["diag"].as<std::string>())) {
+    DLAF_ASSERT(m > 0 && n > 0, m, n);
+    DLAF_ASSERT(mb > 0 && nb > 0, mb, nb);
+
+    if (do_check != dlaf::miniapp::CheckIterFreq::None) {
+      std::cerr << "Warning! At the moment result checking it is not implemented." << std::endl;
+      do_check = dlaf::miniapp::CheckIterFreq::None;
+    }
+  }
+
+  Options(Options&&) = default;
+  Options(const Options&) = default;
+  Options& operator=(Options&&) = default;
+  Options& operator=(const Options&) = default;
+};
+
+template <typename T>
+using matrix_values_t = std::function<T(const GlobalElementIndex&)>;
+
+template <typename T>
+using linear_system_t =
+    std::tuple<matrix_values_t<T>, matrix_values_t<T>, matrix_values_t<T>>;  // A, B, X
+
+template <typename T>
+linear_system_t<T> sampleLeftTr(blas::Uplo uplo, blas::Op op, blas::Diag diag, T alpha, SizeType m);
+}
+
+struct triangularMultiplicationMiniapp {
+  template <dlaf::Backend backend, typename T>
+  static void run(const Options& opts) {
+    Communicator world(MPI_COMM_WORLD);
+    CommunicatorGrid comm_grid(world, opts.grid_rows, opts.grid_cols, Ordering::ColumnMajor);
+
+    // Allocate memory for the matrices
+    dlaf::matrix::Matrix<T, Device::CPU> ah(GlobalElementSize{opts.m, opts.m},
+                                            TileElementSize{opts.mb, opts.mb}, comm_grid);
+    dlaf::matrix::Matrix<T, Device::CPU> bh(GlobalElementSize{opts.m, opts.n},
+                                            TileElementSize{opts.mb, opts.nb}, comm_grid);
+
+    dlaf::matrix::MatrixMirror<T, DefaultDevice_v<backend>, Device::CPU> a(ah);
+    dlaf::matrix::MatrixMirror<T, DefaultDevice_v<backend>, Device::CPU> b(bh);
+
+    auto sync_barrier = [&]() {
+      a.get().waitLocalTiles();
+      b.get().waitLocalTiles();
+      DLAF_MPI_CHECK_ERROR(MPI_Barrier(world));
+    };
+
+    const auto side = opts.side;
+    const auto uplo = opts.uplo;
+    const auto op = opts.op;
+    const auto diag = opts.diag;
+    const T alpha = 2.0;
+
+    double m = ah.size().rows();
+    double n = bh.size().cols();
+    auto add_mul = n * m * m / 2;
+    const double total_ops = dlaf::total_ops<T>(add_mul, add_mul);
+
+    auto [ref_op_a, ref_b, ref_x] = ::sampleLeftTr(uplo, op, diag, alpha, ah.size().rows());
+
+    for (int64_t run_index = -opts.nwarmups; run_index < opts.nruns; ++run_index) {
+      if (0 == world.rank() && run_index >= 0)
+        std::cout << "[" << run_index << "]" << std::endl;
+
+      // setup matrix A and b
+      using dlaf::matrix::util::set;
+      set(ah, ref_op_a, op);
+      set(bh, ref_b);
```

Comment:
doesn't make much difference as we disabled the checks, but it should be
```suggestion
      set(bh, ref_x);
```

Reply:
I did miss this comment when going through the code. I think the naming of the variables is rather confusing if this is the correct one, since `ref_b` is also defined (I needed to go all the way through `sampleLeftTr` to realise what's going on...).